packages/hardhat-ethers/src/helpers.ts

```diff
@@ -23,11 +25,23 @@ export interface FactoryOptions {
 
 const pluginName = "hardhat-ethers";
 
-export async function getSigners(hre: HardhatRuntimeEnvironment) {
+export async function getSigners(
+  hre: HardhatRuntimeEnvironment
+): Promise<SignerWithAddress[]> {
+  const { SignerWithAddress: SignerWithAddressImpl } = await import(
```

Comment:
I hate the `no-shadowed-variable` rule.

Let me know if you'd prefer to import this without aliasing and to import the type as `SignerWithAddressT` or something like that.

Reply:
lgtm 🤷‍♂️

Reply:
the rule is pretty annoying, i know